elements/pfe-switch/src/pfe-switch.scss

```diff
@@ -0,0 +1,189 @@
+// Please see the pfe-sass README for guidance on using these tools
+@import "../../pfe-sass/pfe-sass";
+
+
+$LOCAL: switch;
+
+// Setting all toggle sizing vars
+// will be converted to em in styles so the graphic elments scale with type
+$font-size:             pfe-var(font-size);
+$toggle-width:          42px;
+$toggle-knob-diameter:  14px;
+$toggle-knob-offset:    5px;
+
+// Variables based on PF Next Switch
+$variables: (
+  toggle-knob: (
+    BackgroundColor: pfe-color(surface--lightest),
+    BackgroundColor--disabled: pfe-color(ui-disabled),
+    BoxShadow: '0 0.0625rem 0.0625rem 0rem rgba(3, 3, 3, 0.05), 0 0.25rem 0.5rem 0.25rem rgba(3, 3, 3, 0.06)',
+  ),
+  toggle: (
+    BackgroundColor: #d2d2d2, // @todo - didn't find a great color to match PF4
+    BackgroundColor--checked: pfe-color(ui-base),
+    BackgroundColor--disabled: pfe-color(ui-disabled--text),
+  ),
+  label: (
+    Color--default: pfe-color(text),
+    Color--checked: pfe-color(ui-disabled--text),
+    Color--disabled: pfe-color(ui-disabled--text),
+    Outline--focused: pfe-color(ui-base),
+  ),
+);
+
+:host {
+  @include pfe-set-local($variables);
+  position: relative;
+  display: inline-block;
+  margin: 0 1.5em 0 0;
+  // To hide any negative indented text
+  overflow: hidden;
+}
+
+:host([hidden]) {
+  display: none;
+}
+
+:host(.pfe-focused) {
+  outline: 2px solid #{pfe-local(Outline--focused, $region: label)};
+  outline-offset: 0.25em;
+}
+
+::slotted(input) {
+  // Completely hides the element
+  @extend %sr-only;
+}
+
+::slotted(.pfe-sr-only),
+.pfe-sr-only {
+  // Only hides the text
+  text-indent: -9999em;
+}
+
+::slotted(label) {
+  position: relative;
+  display: block;
+  // In case of parent dom styles, make sure we have enough height
+  min-height: calc(#{pfe-var(line-height)} * 1em);
+  // Width of the switch + some white space
+  padding-left: #{pfe-size-pem($toggle-width) + 0.5em};
+  font-size: $font-size;
+  line-height: pfe-var(line-height);
+  color: #{pfe-local(Color--default, $region: label)};
+  cursor: pointer;
+
+  // Toggle container
+  &:before {
```

Comment:
Unfortunately, this type of selector isn't supported in Safari.
::before and ::after selectors inside ::slotted don't work.
https://github.com/w3c/webcomponents/issues/655

![image](https://user-images.githubusercontent.com/330256/76907003-91868600-687b-11ea-84ac-85a019a16292.png)

Reply:
We'll need to come up with an alternative approach. I'm going to investigate moving light DOM content to the shadow DOM.